File: LeetCode-Java/Find Minimum in Rotated Sorted Array.java
public class Solution {
    public int findMin(int[] num) {
        int start = 0, end = num.length - 1;
        if(num[start] < num[end]) return num[start];
        while(end - start > 1)
        {
            int mid = (start + end) / 2;
            if(num[mid] > num[start]) start = mid;
            else end = mid;
        }
        return num[end];
    }
}

File: Leetcode-cpp/find_minimum_in_rotated_sorted_array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        return findMin(nums, 0, nums.size() - 1);
    }
    
    int findMin(vector<int>& nums, int startIndex, int endIndex) {
        if (nums[startIndex] <= nums[endIndex]) return nums[startIndex];
        int midIndex = (startIndex + endIndex) / 2;
        if (nums[midIndex] >= nums[startIndex])
            return findMin(nums, midIndex + 1, endIndex);
        else
            return findMin(nums, startIndex, midIndex);
    }
};